cacerts/build.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cacerts
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/buildpacks/libcnb"
+
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type Build struct {
+	Logger bard.Logger
+}
+
+// Build returns a libcnb.BuildResult for the given context. Build always contributes a launch layer containing the
+// ca-cert-helper executable.
+//
+// If the buildpack plan contains an entry with name "ca-certificates" Build will contribute a build layer
+// that adds the ca certificates at the paths provided in the plan entry metadata to the system truststore.
+func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
+	var result libcnb.BuildResult
+
+	b.Logger.Title(context.Buildpack)
+
+	var certPaths []string
+	var contributeHelper bool
+	for _, e := range context.Plan.Entries {
+		switch strings.ToLower(e.Name) {
+		case PlanEntryCACertificates:
+			paths, err := pathsFromEntryMetadata(e.Metadata)
+			if err != nil {
+				return libcnb.BuildResult{}, fmt.Errorf("failed to decode CA certificate paths from plan entry:\n%w", err)
+			}
+			certPaths = append(certPaths, paths...)
+		case PlanEntryCACertHelper:
+			contributeHelper = true
```

Comment:
Why not cut out the middle-person?

```go
var certPaths []string
for _, e := range context.Plan.Entries {
	switch strings.ToLower(e.Name) {
	case PlanEntryCACertificates:
		paths, err := pathsFromEntryMetadata(e.Metadata)
		if err != nil {
			return libcnb.BuildResult{}, fmt.Errorf("failed to decode CA certificate paths from plan entry:\n%w", err)
		}
		certPaths = append(certPaths, paths...)
	case PlanEntryCACertHelper:
		h := libpak.NewHelperLayerContributor(context.Buildpack, &context.Plan, "ca-cert-helper", )
		h.Logger = b.Logger
		result.Layers = append(result.Layers, h)
	default:
		return libcnb.BuildResult{}, fmt.Errorf("received unexpected buildpack plan entry %q", e.Name)
	}
}

if len(certPaths) > 0 {
	sort.Strings(certPaths)
	layer := NewTrustedCAs(certPaths)
	layer.Logger = b.Logger
	result.Layers = append(result.Layers, layer)
}

return result, nil
```

Reply:
I didn't want to add more than one helper layer in the (admitted very edge) case that we got two `PlanEntryCACertHelper`s in the buildpack plan.